src/vel_sender.cpp

```diff
@@ -0,0 +1,57 @@
+#include <ros/ros.h>
+#include <cmath>
+#include "lidarPoint.h"
+#include <string>
+#include <iostream>
+#include <geometry_msgs/Twist.h>
+#include <geometry_msgs/Vector3.h>
+
+//TO DO: make callback use the appropriate message type for /prop_local_coords topic
+//
+
+class VelSender {
+public:
+    VelSender() : nh_(""), private_nh_("~")
+    {
+        prop_local_coords_sub_ = nh_.subscribe("/prop_local_coords", 1, &VelSender::velCallback, this);
+        setpoint_velocity_pub_ = nh_.advertise<geometry_msgs::Twist>("/mavros/setpoint_velocity/cmd_vel_unstamped", 1);
+
+        //load parameter for scaling velocity. 1.0 is the default value 
+        private_nh_.param<double>("vel_scale_factor", vel_scale_factor, 1.0); 
+    }
+
+    void spin() {
+        ros::Rate rate(10); // 10 Hz
+        while (ros::ok()) {
+            ros::spinOnce();
+            rate.sleep();
+        }
+    }
+
+private:
+
+    void velCallback(const geometry_msgs::Vector3::ConstPtr& vector_msg)
+    {
+        geometry_msgs::Twist twist_msg;
+        //geometry_msgs::Vector3 vector_msg = *msg;
+        twist_msg.linear.x = vel_scale_factor * vector_msg->x;
```

Comment:
Maybe make this a parameter and put it in params.yaml?

Reply:
that is the way it is, in fact

Reply:
whoops it got removed somehow, just added it back to the params.yaml